### PR TITLE
refactor(ui): Add Icon support to Button component

### PR DESCRIPTION
**Pull request by ckj**
This enables us to pass an `icon` prop to `Button`

Looks like:

screen shot 2018-02-20 at 2 39 57 pm

I've converted all of the 'Add/Create` buttons to utilize it as a first pass.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/buttons/button.jsx [hunk 1]
```diff
@@ -89,12 +102,15 @@ class Button extends React.Component {
...
-    // This container is useless now, but leaves room for when we need to add
-    // components (i.e. icons, busy indicator, etc)
     let childContainer = (
-      <FlowLayout truncate={false}>
-        <span className="button-label">{children}</span>
-      </FlowLayout>
+      <Flex align="center" className="button-label">
+        {icon && (
+          <Icon size={size}>
+            <InlineSvg src={icon} size={size === 'small' ? '12px' : '16px'} />
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/buttons/button.jsx]: future: I think eventually we'll want icons to be a collection of components and to have them all in storybook would be awesome.

File: src/sentry/static/sentry/app/components/buttons/button.jsx [hunk 2]
```diff
@@ -1,12 +1,23 @@
...
-
+import InlineSvg from '../inlineSvg';
 import '../../../less/components/button.less';
 
+const Icon = styled(Box)`
+  margin-right: ${p => (p.size ? '6px' : '8px')};
+  margin-left: -2px;
+
+  /* Need this to get proper vertical alignment */
+  svg {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/buttons/button.jsx]: can we use component selectors here? e.g. `${InlineSvg} {`?

**Review by billyvg — APPROVED** [review]

File: src/sentry/static/sentry/app/components/buttons/button.jsx (hunk 1 above)
Comment (ckj) [reply src/sentry/static/sentry/app/components/buttons/button.jsx]: Agreeeed. Maybe this is something @Chrissy might be interesting in taking on?

**Comment by ckj** [comment]
Closing in favor of #7579.

**Pull request closed without merging**